### PR TITLE
create predefined agent groups when starting server

### PR DESCRIPTION
**Pull request by elyousfi5**

### CODE REVIEW

File: src/ostorlab/runtimes/local/models/models.py [hunk 1]
```diff
@@ -595,10 +596,43 @@ def create_from_agent_group_definition(
...
+        agent_groups = []
+        agent_groups_path = pathlib.Path(agent_groups_path)
+        for agent_group_file in agent_groups_path.iterdir():
+            if agent_group_file.is_file() and agent_group_file.suffix == ".yaml":
+                with open(agent_group_file, "r") as file:
+                    agent_group_definition = definitions.AgentGroupDefinition.from_yaml(
+                        file
+                    )
+                    asset_type = agent_group_file.stem
+                    agent_group = AgentGroup.create_from_agent_group_definition(
```
Comment (rabsondev) [inline src/ostorlab/runtimes/local/models/models.py:629]: Does this check if the agent group exists or it always creates?

File: src/ostorlab/runtimes/local/models/models.py [hunk 2]
```diff
@@ -595,10 +596,43 @@ def create_from_agent_group_definition(
...
+            agent_groups_path: Path to the agent groups folder.
+
+        Returns:
+            List of agent groups.
+        """
+
+        agent_groups = []
+        agent_groups_path = pathlib.Path(agent_groups_path)
+        for agent_group_file in agent_groups_path.iterdir():
+            if agent_group_file.is_file() and agent_group_file.suffix == ".yaml":
```
Comment (oussamaessaji) [inline src/ostorlab/runtimes/local/models/models.py:623]:
```suggestion
            if agent_group_file.is_file() is True and agent_group_file.suffix == ".yaml":
```

File: tests/cli/serve/serve_test.py [hunk 3]
```diff
@@ -36,3 +37,233 @@ def testOstorlabServe_whenRefreshApiKeyFlagIsNotProvided_getOrCreateApiKey(
 
     assert mocked_refresh_api_key.called is False
     assert "API key refreshed" not in result.output
+
+
+def testOstorlabServe_whenStarting_shouldPersistPredifinedAgentGroups(
```
Comment (ErebusZ) [inline tests/cli/serve/serve_test.py:42]: Please split this to have one for each agent group

Comment (elyousfi5) [reply tests/cli/serve/serve_test.py:42]: it doesnt make any sense to do that !

File: src/ostorlab/runtimes/local/models/models.py (hunk 1 above)
Comment (elyousfi5) [reply src/ostorlab/runtimes/local/models/models.py:627]: for now they are always created, to check

File: src/ostorlab/serve_app/app.py [hunk 4]
```diff
@@ -25,6 +26,8 @@ def create_app(path: str = "/graphql", **kwargs) -> flask.Flask:
         ),
     )
 
+    models.AgentGroup.create_from_files(pathlib.Path(__file__).parent / "agent_groups")
```
Comment (3asm) [inline src/ostorlab/serve_app/app.py:29]: create_from_directory

**Review by 3asm — APPROVED** [review]

File: src/ostorlab/serve_app/agent_groups/network.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        type: array
+        description: List of scripts to run using Nmap
+        value:
+          - "banner"
+  - key: agent/ostorlab/asteroid
+  - key: agent/ostorlab/metasploit
+  - key: agent/ostorlab/openvas
+  - key: agent/ostorlab/nuclei
+  - key: agent/ostorlab/tsunami
+    port_mapping: []
```
Comment (deadly-panda) [inline src/ostorlab/serve_app/agent_groups/network.yaml:29]:
```suggestion
```

File: src/ostorlab/serve_app/agent_groups/sbom.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+kind: AgentGroup
+description: SBOM scan
+name: sbom
+agents:
+  - key: agent/ostorlab/osv
+    args:
+      - name: nvd_api_key
+        type: string
+        description: NVD api key.
+        value: ""
```
Comment (deadly-panda) [inline src/ostorlab/serve_app/agent_groups/sbom.yaml:10]: This won't work.
The value should not be set. if it's an empty string, the `apiKey` header will be set to an empty string and the request to the nvd-api call will fail & keep retrying.

**Review by deadly-panda — CHANGES_REQUESTED** [review]

File: src/ostorlab/serve_app/agent_groups/sbom.yaml (hunk 6 above)
Comment (elyousfi5) [reply src/ostorlab/serve_app/agent_groups/sbom.yaml]: addressed

**Review by 3asm — APPROVED** [review]

**Pull request merged**